jfix-armeria-facade/src/main/kotlin/ru/fix/armeria/facade/webclient/impl/BaseHttpClientBuilderImpl.kt

```diff
@@ -158,19 +159,25 @@ internal abstract class BaseHttpClientBuilderImpl<out HttpClientBuilderT : BaseH
             else -> WebClient.builder()
         }
 
-        // build ClientFactory
-        var clientFactoryBuilder: ClientFactoryBuilder = baseBuilderState.clientFactoryBuilder()
-        clientFactoryBuilder = clientFactoryBuilder.workerGroup(
-            EventLoopGroups.newEventLoopGroup(
-                baseBuilderState.ioThreadsCount ?: Flags.numCommonWorkers(),
-                "$clientName-eventloop"
-            ),
-            true
-        )
-        val clientFactoryBuilderCustomizer = baseBuilderState.clientFactoryBuilderCustomizer
-        val clientFactory = clientFactoryBuilder
-            .clientFactoryBuilderCustomizer()
-            .build()
+        val clientFactory: ClientFactory
+        if (baseBuilderState.clientFactory == null) {
+            // build ClientFactory
+            var clientFactoryBuilder: ClientFactoryBuilder = baseBuilderState.clientFactoryBuilder()
+            clientFactoryBuilder = clientFactoryBuilder.workerGroup(
+                EventLoopGroups.newEventLoopGroup(
+                    baseBuilderState.ioThreadsCount ?: Flags.numCommonWorkers(),
+                    "$clientName-eventloop"
+                ),
+                true
+            )
+            val clientFactoryBuilderCustomizer = baseBuilderState.clientFactoryBuilderCustomizer
+            clientFactory = clientFactoryBuilder
+                .clientFactoryBuilderCustomizer()
+                .build()
+        } else {
+            // use passed ClientFactory
+            clientFactory = baseBuilderState.clientFactory
+        }
```

Comment:
we can rewrite it in a more kotlin-friendly way :)

```suggestion
        val clientFactory: ClientFactory = if (baseBuilderState.clientFactory == null) {
            // build ClientFactory
            var clientFactoryBuilder: ClientFactoryBuilder = baseBuilderState.clientFactoryBuilder()
            clientFactoryBuilder = clientFactoryBuilder.workerGroup(
                EventLoopGroups.newEventLoopGroup(
                    baseBuilderState.ioThreadsCount ?: Flags.numCommonWorkers(),
                    "$clientName-eventloop"
                ),
                true
            )
            val clientFactoryBuilderCustomizer = baseBuilderState.clientFactoryBuilderCustomizer
            clientFactoryBuilder
                .clientFactoryBuilderCustomizer()
                .build()
        } else {
            // use passed ClientFactory
            baseBuilderState.clientFactory
        }
```

Reply:
fixed